src/backend/utils/adt/gp_optimizer_functions.c

```diff
@@ -16,47 +16,78 @@
 #include "funcapi.h"
 #include "utils/builtins.h"
 
-extern Datum EnableXform(PG_FUNCTION_ARGS);
+typedef Datum (*gp_opt_version_func) (void);
+
+/*
+ * A stub function to call if no optimizer function is found.
+ */
+static Datum
+gp_optimizer_function_stub(PG_FUNCTION_ARGS)
+{
+	return CStringGetTextDatum("Server has been compiled without ORCA");
+}
+
+/*
+ * Loads optimizer function from a shared library. If library is not presented
+ * or doesn't contain the requested function or any of its dependencies,
+ * returns the stub function.
+ */
+static PGFunction
+gp_optimizer_load_function(char *funcname)
+{
+	volatile PGFunction func = NULL;
```

Comment:
why not
```suggestion
	volatile PGFunction func = gp_optimizer_function_stub;
```

Reply:
`load_external_function()` can return NULL without throwing an error, so anyway we need to check for NULL value after the catch block. Thus no need to assign the stub at the beginning.